cmd/frontend/internal/app/ui/handlers.go

```diff
@@ -256,6 +257,75 @@ func serveSignIn(w http.ResponseWriter, r *http.Request) error {
 	return renderTemplate(w, "app.html", common)
 }
 
+// redirectTreeOrBlob redirects a blob page to a tree page if the file is actually a directory,
+// or a tree page to a blob page if the directory is actually a file.
+func redirectTreeOrBlob(routeName string, common *Common, r *http.Request, w http.ResponseWriter) (requestHandled bool, err error) {
```

Comment:
nit: It is common to have `http.ResponseWriter` comes before `*http.Request`. I don't know why but everybody seems using this order because the stdlib `net/http.HandlerFunc` uses it.
```suggestion
func redirectTreeOrBlob(routeName string, common *Common, w http.ResponseWriter, r *http.Request) (requestHandled bool, err error) {
```